semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -1036,6 +1036,10 @@ let rec stmt_aux env st =
         | _ -> (None, manager_as_pat)
       in
       python_with_stmt env manager opt_pat body
+  (* Java: synchronized (E) S *)
+  | G.OtherStmt (G.OS_Sync, [ G.E objorig; G.S stmt1 ]) ->
```

Comment:
hmm, it should be a OtherStmtWithStmt then. We should remove G.OS_Sync and move it under OtherStmtWithStmt.

Reply:
I will open a separate PR for that.